### PR TITLE
Azure Identity Spring Helper

### PR DESCRIPTION
**Pull request by mnriem**

### CODE REVIEW

File: sdk/spring/azure-identity-spring-library/src/main/java/com/microsoft/azure/identity/spring/AzureIdentitySpringHelper.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        
+        String tenantIdKey = AZURE_CREDENTIAL_PREFIX + name + "tenantId";
+        String clientIdKey = AZURE_CREDENTIAL_PREFIX + name + "clientId";
+        String clientSecretKey = AZURE_CREDENTIAL_PREFIX + name + "clientSecret";
+
+        String tenantId = environment.getProperty(tenantIdKey);
+        String clientId = environment.getProperty(clientIdKey);
+        String clientSecret = environment.getProperty(clientSecretKey);
+
+        if (tenantId != null && clientId != null && clientSecret != null) {
```
Comment (yevster) [inline sdk/spring/azure-identity-spring-library/src/main/java/com/microsoft/azure/identity/spring/AzureIdentitySpringHelper.java:106]: If some but not all of these are none, that would suggest an unintentional error condition. Perhaps it's worth failing loudly.

File: sdk/spring/azure-identity-spring-library/pom.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+    </properties>
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+            <version>5.2.8.RELEASE</version>
```
Comment (yevster) [inline sdk/spring/azure-identity-spring-library/pom.xml:23]: I'd make this a property shared between this artifact and the next.

Comment (yevster) [reply sdk/spring/azure-identity-spring-library/pom.xml:23]: Actually, wouldn't this version be specified in the parent POM anyway?

File: sdk/spring/azure-identity-spring-library/src/main/java/com/microsoft/azure/identity/spring/AzureIdentitySpringHelper.java (hunk 1 above)
Comment (mnriem) [reply sdk/spring/azure-identity-spring-library/src/main/java/com/microsoft/azure/identity/spring/AzureIdentitySpringHelper.java:148]: Done

File: sdk/spring/azure-identity-spring-library/pom.xml (hunk 2 above)
Comment (mnriem) [reply sdk/spring/azure-identity-spring-library/pom.xml:23]: In the other POMs of the sdk/spring sub project they are all in there too. I'll let others decide what needs to be done

**Comment by jialindai** [comment]
If the library will also be a spring starter, we should give it a name following the guideline (https://github.com/Azure/azure-sdk/blob/master/docs/java/spring.md)

I'd like to suggest change the artifact id to: azure-spring-boot-starter-identity

For the implementation part, we should provide an API looks like azure-identity API. So I am thinking to provide SpringTokenCredential and SpringTokenCredentialBuilder which follows the ChainedCredential pattern. The builder should be able to accept different property prefix.

When the API is ready, a default SpringTokenCredential (property prefix azure.credential) can be injected into Spring context. If users want additional token credential based on different property prefix, the API allows them to do it easily in configuration class.

**Comment by yevster** [comment]
> If the library will also be a spring starter, we should give it a name following the guideline (https://github.com/Azure/azure-sdk/blob/master/docs/java/spring.md)
> 
> I'd like to suggest change the artifact id to: azure-spring-boot-starter-identity
> 
> For the implementation part, we should provide an API looks like azure-identity API. So I am thinking to provide SpringTokenCredential and SpringTokenCredentialBuilder which follows the ChainedCredential pattern. The builder should be able to accept different property prefix.
> 
> When the API is ready, a default SpringTokenCredential (property prefix azure.credential) can be injected into Spring context. If users want additional token credential based on different property prefix, the API allows them to do it easily in configuration class.
> 
> @seanli1988 for his information.

I'm not sure what adding `Spring` to all these classes would accomplish. There's no connection between Azure token credentials and Spring. There's nothing Spring-specific to them. So I can't imagine what purpose unnecessary Spring-specific classes would serve at the expense of creating customer confusion.

**Comment by mnriem** [comment]
I have to agree with @yevster on this one since there is no Spring abstraction we are implementing we should just let TokenCredential be what we expose.

**Comment by stliu** [comment]
is this still active? @mnriem

**Comment by mnriem** [comment]
I don't think so

**Comment by stliu** [comment]
okay, i'm closing this one , please reopen a new one if needed

**Pull request closed without merging**